dashboard/test/controllers/courses_controller_test.rb

```diff
@@ -7,35 +7,123 @@ class CoursesControllerTest < ActionController::TestCase
     @teacher = create :teacher
     sign_in @teacher
 
+    @levelbuilder = create :levelbuilder
+
     plc_course = create :plc_course, name: 'My Plc'
     @course_plc = plc_course.course
     @course_regular = create :course, name: 'non-plc-course'
+
+    # stub writes so that we dont actually make updates to filesystem
+    File.stubs(:write)
   end
 
-  test "plc courses get sent to user_course_enrollments_controller" do
+  # Tests for show
+
+  test "show: plc courses get sent to user_course_enrollments_controller" do
     get :show, params: {course_name: @course_plc.name}
     assert_template 'plc/user_course_enrollments/index'
   end
 
-  test "plc course names get titleized" do
+  test "show: plc course names get titleized" do
     get :show, params: {course_name: 'my_plc'}
     assert_template 'plc/user_course_enrollments/index'
   end
 
-  test "regular courses get sent to show" do
+  test "show: regular courses get sent to show" do
     get :show, params: {course_name: @course_regular.name}
     assert_template 'courses/show'
   end
 
-  test "regular courses do not get titlized" do
+  test "show: regular courses do not get titlized" do
     assert_raises ActiveRecord::RecordNotFound do
       get :show, params: {course_name: 'non_plc_course'}
     end
   end
 
-  test "non existant course throws" do
+  test "show: non existant course throws" do
     assert_raises ActiveRecord::RecordNotFound do
       get :show, params: {course_name: 'nosuchcourse'}
     end
   end
+
+  # Tests for create
+
+  test "create: fails without levelbuilder permission" do
+    Rails.application.config.stubs(:levelbuilder_mode).returns true
```

Comment:
This makes me wish we had individual test blocks where we could run code before each 😞.  For now, I wonder if this is a sign it should live in a separate file, with it's own setup step?  I think the answer is probably no, since we'd have to duplicate the setup from this file (or extract it somewhere it could be re-used)...